File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/drivers/interfaces/SecureServer.java
package de.fhg.aisec.ids.idscp2.drivers.interfaces;

/**
 * An interface for the IDSCPv2 Secure Server
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
public interface SecureServer {

    /*
     * Terminate the secure server
     */
    void safeStop();

    /*
     * Check if the secure server is running
     */
    boolean isRunning();
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/drivers/interfaces/SecureServer.kt
package de.fhg.aisec.ids.idscp2.drivers.interfaces

/**
 * An interface for the IDSCP2 Secure Server
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
interface SecureServer {
    /*
     * Terminate the secure server
     */
    fun safeStop()

    /*
     * Check if the secure server is running
     */
    val isRunning: Boolean
}